frontend/src/components/Pages/FrontPage/MissionOverview/MissionControlButtons.tsx

```diff
@@ -27,12 +27,12 @@ const ButtonText = styled.div`
 export function MissionControlButtons({ mission }: MissionProps) {
     const { TranslateText } = useLanguageContext()
     const { missionControlState, updateMissionState } = useMissionControlContext()
+    const missionStatus = mission.status
 
-    const renderControlIcon = (missionStatus: MissionStatus) => {
-        if (missionControlState.isWaitingForResponse) {
-            return <CircularProgress size={32} />
-        } else if (missionStatus === MissionStatus.Ongoing) {
-            return (
+    return (
+        <>
+            {missionControlState.isWaitingForResponse && <CircularProgress size={32} />}
+            {missionStatus === MissionStatus.Ongoing && (
```

Comment:
Maybe use mission.status directly?

```suggestion
    return (
        <>
            {missionControlState.isWaitingForResponse && <CircularProgress size={32} />}
            {mission.status === MissionStatus.Ongoing && (
```